Add text on GPU compute model (#105)
darchr/gem5-20-paper: changes/gpu-compute-model.tex

\subsection[GPU Compute Model]{GPU Compute Model\footnote{by Anthony Gutierrez}}
\label{sec:gpu}

GPUs have become an important part of the system design for high-performance
computing, machine learning, and many other workloads. Thus, we have integrated
a compute-based GPU model into gem5~\cite{GutierrezBeckmann2018-amdAPU}. The
GPU model is based on AMD's Graphics Core Next (GCN) architecture

> \footnote{http://amd.com/en/technologies/gcn}, and at the time of this writing

Reviewer: ```suggestion
- at the time of this writing
+ currently
```

(I can also submit a separate pull request for this if it requires you going through legal again)

Reviewer: Sorry, the suggested edit did not show up properly.  It was just replacing "at the time of this writing" with "currently", not removing the footnote.

Author: It's fine, I understood. Done.

the SIMD units, each CU can have some arbitrary number of scalar units, scratch
pad memory, and global memory units (both vector and scalar memory).

The SIMD unit relies on large vector register files (VRF), however the GPU can
also be configured to include a scalar register file (SRF) for use with scalar
execution units. Additionally, vector ALU and memory instructions may contain
scalar operands, and thus vector operations require access to the SRF.

At a high level the CU pipeline modeled in gem5 consists of four stages: \textit{fetch},
\textit{scoreboard check}, \textit{schedule}, and \textit{execute}. The fetch stage
is responsible for arbitrating instruction fetch, storing fetched data,
and decoding instructions for all wavefronts on the CU. The scoreboard check stage is
responsible for ensuring that intra-wavefront dependences are
correctly handled before instructions can be scheduled. The schedule stage is
responsible for arbitrating among the wavefronts presented by
the scoreboard check stage for each execution unit type. Once an instruction is dispatched
to an execution unit it is guaranteed to execute.

> This "execute at execute" approach is the same as used by the gem5 in-order and

Reviewer 2: ```suggestion
This ``execute at execute'' approach is the same as used by the gem5 in-order and
```

As it is, this won't render correctly when compiled.

Author: Done

> O3 cpu models. The execute stage executes the instruction by calling the \textit{execute}

Reviewer 2: ```suggestion
out-of-order CPU models. The execute stage executes the instruction by calling the \textit{execute}
```

Author: Done
